### PR TITLE
Fix Monaco bugs

### LINKED ISSUE #769: Run keyboard shortcut does not work under Monaco
As reported [here](https://www.reddit.com/r/rust/comments/s462cp/rust_playground_now_supports_monaco_editor/hsqzxrp/?context=3) `Ctrl+Enter` (Windows) `Cmd+Enter` (Mac) just act as `Enter` under Monaco, instead of activating run/build/test/etc.

### LINKED ISSUE #770: Monaco editor does not capture focus
With the ace editor, when you load the playground, the focus is already in the code box, so you can just start typing. With the monaco editor, the focus is not captured, and you first have to click in the box or tab to it (10 times in my testing) in order to get to start typing code.

### LINKED ISSUE #771: Monaco editor does not handle raw string literals properly in its syntax highlighting
Here's some sample code:
```rust
    let schema = 
r#"
name: HelloWorld

events:
  B: {}
  F: {}
  A: {}
  C: {}
  D: {}
  E: {}
  G: {}
"#;
```
It looks like this in the monaco editor:
image
and like this in the ace editor:
image

### PR DESCRIPTION
**Pull request by HKalbasi**
fix #771
fix #770 
fix #769

### CODE REVIEW

File: ui/frontend/webpack.config.js [hunk 1]
```diff
@@ -158,7 +158,7 @@ module.exports = function(_, argv) {
       }),
       new MonacoWebpackPlugin({
         filename: `${filenameTemplate}.worker.js`,
-        languages: ['rust'],
+        languages: [],
```
Comment (shepmaster) [inline ui/frontend/webpack.config.js:161]: Why this change?

File: ui/frontend/editor/MonacoEditorCore.tsx [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-import MonacoEditor, { EditorWillMount } from 'react-monaco-editor';
+import MonacoEditor, { EditorDidMount, EditorWillMount } from 'react-monaco-editor';
 import { useSelector } from 'react-redux';
 import State from '../state';
 import { config, grammar, themeVsDarkPlus } from './rust_monaco_def';
 
 import styles from './Editor.module.css';
 
-const MODE_ID = 'my-rust';
+const MODE_ID = 'rust';
```
Comment (shepmaster) [inline ui/frontend/editor/MonacoEditorCore.tsx:10]: Why this change?

File: ui/frontend/webpack.config.js (hunk 1 above)
Comment (HKalbasi) [reply ui/frontend/webpack.config.js:161]: We don't need their grammar because we use our own anyway. It should save some bits.

File: ui/frontend/editor/MonacoEditorCore.tsx (hunk 2 above)
Comment (HKalbasi) [reply ui/frontend/editor/MonacoEditorCore.tsx:10]: Somethings (like hover snippets) can not be set to use `my-rust`. Not related to the issues, but still something that isn't correct.

**Comment by shepmaster** [comment]
In the future, please create separate commits (or separate PRs if applicable) and use the commit message to describe what the commit is doing and why.

File: ui/frontend/editor/rust_monaco_def.ts [hunk 3]
```diff
@@ -71,6 +71,9 @@ export const grammar: languages.IMonarchLanguage = {
   // The main tokenizer for our languages
   tokenizer: {
     root: [
+      [/r"/, { token: 'string.quote', next: '@rawstring0' }],
+      [/r#"/, { token: 'string.quote', next: '@rawstring1' }],
+      [/r##"/, { token: 'string.quote', next: '@rawstring2' }],
```
Comment (alexschrod) [inline ui/frontend/editor/rust_monaco_def.ts:76]: Won't doing it this way only support raw strings with up to two `#`s? The ace editor supports an unlimited amount of `#`s, and I'd be amazed if this Monaco tokenizer engine wasn't built to offer up a way to match the starting `#`s to the ending `#`s.

Comment (alexschrod) [reply ui/frontend/editor/rust_monaco_def.ts:76]: This ought to do it (see also second suggestion below):
```suggestion
      [/r"/, { token: 'string.quote', next: '@rawstring0' }],
      [/r(#+)"/, { token: 'string.quote', next: '@rawstring1.$1' }],
```

File: ui/frontend/editor/rust_monaco_def.ts [hunk 4]
```diff
@@ -130,6 +133,10 @@ export const grammar: languages.IMonarchLanguage = {
       [/[\/*]/, 'comment'],
     ],
 
+    rawstring0: [[/[^"]+/, 'string'], [/"/, { token: 'string.quote', next: '@pop' }]],
+    rawstring1: [[/[^"#]+/, 'string'], [/"#/, { token: 'string.quote', next: '@pop' }]],
+    rawstring2: [[/[^"##]+/, 'string'], [/"##/, { token: 'string.quote', next: '@pop' }]],
```
Comment (alexschrod) [inline ui/frontend/editor/rust_monaco_def.ts:138]:
```suggestion
    rawstring0: [[/[^"]+/, 'string'], [/"/, { token: 'string.quote', next: '@pop' }]],
    rawstring1: [
      [/"(#+)/, {
        cases: {
          '$1==$S2': { token: 'string.quote', next: '@pop' },
          '@default': { token: 'string' }
        },
      }],
      [/./, 'string'],
    ]
```

**Review by alexschrod — CHANGES_REQUESTED** [review]
This should allow an unlimited amount of `#` usage in raw strings.

File: ui/frontend/editor/rust_monaco_def.ts (hunk 3 above)
Comment (HKalbasi) [reply ui/frontend/editor/rust_monaco_def.ts]: I applied your suggestion, thanks!
